### PR TITLE
DOC: add key concepts doc

### PR DESCRIPTION
**Pull request by jakevdp**
This will replace the new content in thinking-in-jax within the new tutorial flow.

Followup to #20306

Rendered version will appear at: https://jax--20347.org.readthedocs.build/en/20347/tutorials/key-concepts.html

### CODE REVIEW

File: docs/glossary.rst [hunk 1]
```diff
@@ -3,6 +3,9 @@ JAX Glossary of Terms
 
 .. glossary::
 
+    Array
+      JAX's analog of the :class:`numpy.ndarray`. See :class:`jax.Array`.
```
Comment (skye) [inline docs/glossary.rst:7]:
```suggestion
      JAX's analog of :class:`numpy.ndarray`. See :class:`jax.Array`.
```

File: docs/tutorials/key-concepts.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(key-concepts)=
+# Key Concepts
+
+This sectiong briefly introduces some key concepts of the JAX package.
```
Comment (skye) [inline docs/tutorials/key-concepts.md:18]:
```suggestion
This section briefly introduces some key concepts of the JAX package.
```

File: docs/tutorials/key-concepts.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+Transformations like {func}`~jax.jit`, {func}`~jax.vmap`, {func}`~jax.grad`, and others are
+key to using JAX effectively, and we'll cover them in detail in later sections.
+
+(key-concepts-tracing)=
+## Tracing
+
+The magic behind transformations is the notion of a {term}`Tracers <Tracer>`.
+Tracers are abstract standins for array objects, and are passed to JAX functions in order
```
Comment (skye) [inline docs/tutorials/key-concepts.md:98]:
```suggestion
Tracers are abstract stand-ins for array objects, and are passed to JAX functions in order
```

**Review by skye — APPROVED** [review]

File: docs/glossary.rst (hunk 1 above)
Comment (jakevdp) [reply docs/glossary.rst:7]: Done

File: docs/tutorials/key-concepts.md (hunk 2 above)
Comment (jakevdp) [reply docs/tutorials/key-concepts.md:18]: Done

File: docs/tutorials/key-concepts.md (hunk 3 above)
Comment (jakevdp) [reply docs/tutorials/key-concepts.md:98]: Done

**Pull request merged**